app/ui/browser/model/index.js

```diff
@@ -29,6 +34,8 @@ export const State = Immutable.Record({
   // This is a list of Page objects (see below)
   pages: Immutable.List(),
 
+  pageIds: Immutable.Set(),
```

Comment:
After going through the code, I still don't quite get why this set exists; wouldn't this denormalizing duplicate what we have in pages, which we could do with `pages.map(p => p.id)`?